### PR TITLE
Dev 1

### PR DESCRIPTION
**Pull request by kar3nmk**

### CODE REVIEW

File: src/components/user/UserEditModal.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+} from 'antd';
+import { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+export function UserEditModal(props) {
+  const { data, edit } = props;
+  const [visible, setVisible] = useState(false);
+  const [values, setValues] = useState(data);
+  const initialValues = useRef(values);
```
Comment (mar-taron) [inline src/components/user/UserEditModal.js]: why we need initialValues as ref?

File: src/containers/user/Filters.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { connect } from 'react-redux';
+import { Filters } from '../../components/user/Filters';
+import { getUsersByFilter } from '../../actions/user/UserActionCreator';
+
+const mapDispatchToProps = (dispatch) => ({
```
Comment (mar-taron) [inline src/containers/user/Filters.js]: use bindActionCreators

File: src/reducers/Settings.reducer.js [hunk 3]
```diff
@@ -5,12 +5,12 @@ const initialState = {
...
 
-export function settings(state = initialState, action) {
-  switch (action.type) {
+export function settings(state = initialState, { type, payload }) {
+  switch (type) {
   case SettingsActionTypes.FAILURE:
     return {
       ...state,
-      messages: Object.values(action.payload),
+      messages: Object.values(payload.errors),
```
Comment (mar-taron) [inline src/reducers/Settings.reducer.js]: no need to write logic here

File: src/components/user/UserEditModal.js (hunk 1 above)
Comment (kar3nmk) [reply src/components/user/UserEditModal.js]: Because in this case, we need(can) to change the state without additional rerendering.

File: .env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+BASE_URL='http://localhost:8000'
```
Comment (mar-taron) [inline .env]: no need quotes here

File: .eslintrc.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ module.exports = {
   settings: {
     'import/resolver': {
       node: {
-        moduleDirectory: ['node_modules', './src'],
+        moduleDirectory: ['node_modules', 'src'],
```
Comment (mar-taron) [inline .eslintrc.js:24]: try with jsconfig

File: src/components/auth/Verify.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { useParams } from 'react-router-dom';
 import { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { VerifyResult } from './VerifyResult';
-import { Spin } from '../settings/Spin';
+import { SpinStyled } from './Auth.styled';
```
Comment (mar-taron) [inline src/components/auth/Verify.js]: Spin Styled should be in global place , it's not only for auth

File: src/components/user/Filters.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+import PropTypes from 'prop-types';
+import { getIsoDate } from 'utils';
+import {
+  FilterInputWrapperStyled,
+  FiltersButtonsWrapper,
+  FiltersDatePickerStyled,
+  FiltersInputStyled,
+  FilterWrapper,
+} from './User.styled';
+import LocalStorageService from '../../services/localStorage.service';
```
Comment (mar-taron) [inline src/components/user/Filters.js]: try not to use relative path

File: src/components/user/Filters.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+export function Filters(props) {
+  const { setFilterParams, setPagination, getUsers } = props;
+  const [form] = Form.useForm();
+
+  const onFinish = (filterParams) => {
+    const requestData = { ...filterParams, createdAt: getIsoDate(filterParams.createdAt) };
+    setPagination((prevState) => ({ total: prevState.total, pageSize: 10, current: 1 }));
+    setFilterParams(requestData);
+    getUsers({ page: 1, limit: 10, ...filterParams });
```
Comment (mar-taron) [inline src/components/user/Filters.js]: you don't need to send limit params, only page, and pageSize, limit is defined in backend part, the same for evereywhere - reset

File: src/components/user/Filters.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+  const onReset = () => {
+    form.resetFields();
+    setFilterParams({});
+    setPagination((prevState) => ({ total: prevState.total, pageSize: 10, current: 1 }));
+    getUsers({ page: 1, limit: 10 });
+  };
+
+  const [showFilter, setShowFilter] = useState(LocalStorageService.get('usersFilterView'));
+  const viewHandler = () => {
```
Comment (mar-taron) [inline src/components/user/Filters.js:35]: no need to keep this in localstorage, but in URL like, expand = true/false

**Review by mar-taron — APPROVED** [review]

**Pull request merged**